src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,41 +1,80 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  todos: Todo[];
+  select: number;
+  setSelect: (select: number) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({ todos, select, setSelect }) => {
+  const [loading, setLoading] = useState(false);
+  const [user, setUser] = useState<User>();
+
+  const selectedTodo = todos.find(todo => todo.id === select);
+  const foundedUserId = selectedTodo?.userId;
+
+  useEffect(() => {
+    setLoading(true);
+
+    if (foundedUserId) {
+      getUser(foundedUserId)
+        .then(userFromServer => setUser(userFromServer))
+        .finally(() => setLoading(false));
+    }
+  }, [foundedUserId]);
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+      {select !== 0 && <div className="modal-background" />}
 
-      {true ? (
+      {loading ? (
         <Loader />
       ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
-
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-            </p>
-          </div>
-        </div>
+        todos.map(
+          todo =>
+            select === todo.id && (
```

Comment:
The condition `select === todo.id` is used to render the modal content. Ensure that the `select` state is correctly managed to reflect the intended selection logic and avoid displaying incorrect data.